### PR TITLE
prov/efa: Use EFA libibverbs provider

### PR DESCRIPTION
**Pull request by rwespetal**
Update the EFA Libfabric provider to use libibverbs.

### CODE REVIEW

**Comment by rajachan** [comment]
The rdma-core in Travis is too old for EFA (v17.x). Until https://github.com/ofiwg/libfabric/pull/5550, the build used to clone and build rdma-core, we're going to have to go back to doing that that (but with a new enough version).

**Comment by jsquyres** [comment]
@rajachan Sadness.  Sorry about that.  😢  The good news is that you can restore the rdma-core download/configure/build/install in the Travis setup out of history from just a few commits ago.

**Comment by rajachan** [comment]
> The good news is that you can restore the rdma-core download/configure/build/install in the Travis setup out of history from just a few commits ago.

I knew this PR was coming back then, but the version in there was old enough already (v13) so figured we'll deal with it when we got here.

File: prov/efa/configure.m4 [hunk 1]
```diff
@@ -31,9 +40,58 @@ AC_DEFUN([FI_EFA_CONFIGURE],[
...
+	      [FI_CHECK_PACKAGE([efadv],
+				[infiniband/efadv.h],
+				[efa],
+				[efadv_query_ah],
+				[-libverbs],
+				[$efa_PREFIX],
+				[$efa_LIBDIR],
+				[efa_happy=1],
+				[efa_happy=0])
+	      ])
```
Comment (wzamazon) [inline prov/efa/configure.m4]: Would you please add a check for function efadv_query_device() too? That function was also used by this patch.

Comment (rwespetal) [reply prov/efa/configure.m4]: Sure, I've added a check for that.

**Comment by rwespetal** [comment]
bot:aws:retest

**Comment by rwespetal** [comment]
I forgot to mention that I added the change to Travis in this PR. It's building rdma-core v27 and has the EFA and verbs providers use that build for the test.

**Comment by rwespetal** [comment]
bot:aws:retest

**Pull request merged**